api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java

```diff
@@ -0,0 +1,80 @@
+package org.pmiops.workbench.billing;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.common.collect.Streams;
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BillingAlertsService {
+
+  private final BigQueryService bigQueryService;
+  private final NotificationService notificationService;
+  private final UserDao userDao;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  @Autowired
+  public BillingAlertsService(
+      BigQueryService bigQueryService,
+      NotificationService notificationService,
+      UserDao userDao,
+      Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.bigQueryService = bigQueryService;
+    this.notificationService = notificationService;
+    this.userDao = userDao;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+  }
+
+  public void alertUsersExceedingFreeTierBilling() {
+    QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(
```

Comment:
Would make sense to have a new Dao class to contain this query?

Reply:
Is the BQ table's cost always aligned with the current billing cycle? I.e. is there a way to run this job on historical data to see how many users exceeded the free tier two months ago?